app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt

```diff
@@ -231,7 +231,7 @@ class UrlInputFragment :
 
     private fun updateTipsLabel() {
         val context = context ?: return
-        val tips = TipManager.getAvailableTips(context)
+        val tips = TipManager().getAvailableTips(context)
```

Comment:
I understand why you've chosen to make this an instance instead of the original singleton (so that we don't leak the context). However, similar to my comment above, I think we would regress in performance if we're doing this work every time we focus on the `UrlInputFragment`.

Why not create a lazy singleton that takes the context as a parameter so you can call `getAvailableTips` safely?


```kotlin
// In Components.kt
val tipManager by lazy { TipManager(context) }

// In UrlInputFragment.kt
val tips = context.components.tipManager.getAvailableTips()
```

Does that work?

Reply:
It's working . I made the changes . Thank you!

Reply:
Excellent, I'm glad it worked out!